### DISCUSSION TITLE
Does fix imbalance change dataset size

### DISCUSSION THREAD

**Discussion by suresh-chinta** [pycaret/pycaret#1224] [Q&A] [answered]
I have set fix_imbalance to True in setup, I do not see it changing the distribution of target, especially it is not increasing the minority sample. Is my observation incorrect? 

Also what is the difference between setting fix_imbalance at the time of setup vs create_model?

When fix_imbalance is set to True in create_model for lightgum as below, what does it do? does it set is_unbalance or scale_pos_weight of lightgbm?
create_model('lightgbm', round = 2, fix_imbalance = True)

Can I directly set lightgbm parameter (is_unbalance) as below:
create_model('lightgbm', round = 2, is_unbalance= True)

Thank you

**Comment by Yard1** [comment, marked as answer]
`fix_imbalance=True` applies SMOTE during cross validation and model fitting (without changing the train dataset).

**Reply by suresh-chinta** [reply]
Thank you.

**Reply by Anagha-github** [reply]
@Yard1 How can i view the data balanced by SMOTE ? I understand, it is not changing my train dataset. Pycaret says it uses SMOTE when fix_imbalance = True, I wanna see the balanced data. Please help.

**Comment by Anagha-github** [comment]
@Yard1 How can i view the data balanced by SMOTE ? I understand, it is not changing my train dataset. Pycaret says it uses SMOTE when fix_imbalance = True, I wanna see the balanced data. Please help.
